### DISCUSSION TITLE
Github Action doesn't find a specific file

### DISCUSSION THREAD

**Discussion by saigkill** [community/community#65165] [Actions] [answered]
### Select Topic Area

Question

### Body

Hello Community,

i have prepared a [repository](https://github.com/saigkill/test/tree/main). It contains two directories, one of them is "Mitgliedsantrag". Inside that directory i have a "main.tex" file.

I'm using that workflow file:

name: Latex Build
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - run: |
            cd Mitgliedsantrag      
      - uses: xu-cheng/latex-action@v2
        with:
          root_file: main.tex
      - run: |
          cp main.pdf Mitgliedsantrag.pdf
      - uses: actions/upload-artifact@v3
        with:
          name: PDF
          path: Mitgliedsantrag.pdf

The "cd Mitgliedsantrag" looks fine and is shown as executed. But the Latex Action doesn't find the main.tex file.

The last job protocol is on [Job](https://github.com/saigkill/test/actions/runs/5993405566/job/16253817749).

Maybe anyone knows, why it doesn't work?

Greetings
Sascha

**Comment by arafajs** [comment, marked as answer]
The reason why the latex-action is not finding the main.tex file is because the cd Mitgliedsantrag step is not changing the working directory of the latex-action. The latex-action by default looks for the main LaTeX file in the current working directory, which is still the root directory of the repository.

To fix this, you can add the following line to the `latex-action` step:

`working-directory: Mitgliedsantrag`

This will tell the latex-action to change to the Mitgliedsantrag directory before trying to find the main LaTeX file.

Here is the updated workflow file:

```
name: Latex Build
on:
push:
branches:
- main
pull_request:
branches:
- main
jobs:
build:
runs-on: ubuntu-latest
steps:
- uses: actions/checkout@v3
- run: |
cd Mitgliedsantrag
- uses: xu-cheng/latex-action@v2
with:
root_file: main.tex
working-directory: Mitgliedsantrag
- run: |
cp main.pdf Mitgliedsantrag.pdf
- uses: actions/upload-artifact@v3
with:
name: PDF
path: Mitgliedsantrag.pdf
```
